routing/routing_tests/route_tests.cpp

```diff
@@ -176,52 +162,84 @@ UNIT_TEST(DistanceToCurrentTurnTest)
     // Initial point.
     auto pos = kTestGeometry[0];
 
-    route.GetCurrentTurn(distance, turn);
+    route.GetNearestTurn(distance, turn);
     size_t currentTurnIndex = 2; // Turn with m_index == 1 is None.
     TEST(base::AlmostEqualAbs(distance,
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = 0;
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
   }
   {
     // Move between points 1 and 2.
     auto pos = (kTestGeometry[1] + kTestGeometry[2]) / 2;
     route.MoveIterator(GetGps(pos.x, pos.y));
 
-    route.GetCurrentTurn(distance, turn);
+    route.GetNearestTurn(distance, turn);
     size_t currentTurnIndex = 2;
     TEST(base::AlmostEqualAbs(distance,
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = (kTestTimes[1 - 1] + kTestTimes[2 - 1]) / 2;
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
   }
   {
     // Move between points 2 and 3.
-    auto pos = (kTestGeometry[2] + kTestGeometry[3]) / 2;
+    auto pos = kTestGeometry[2] * 0.8 + kTestGeometry[3] * 0.2;
     route.MoveIterator(GetGps(pos.x, pos.y));
 
-    route.GetCurrentTurn(distance, turn);
+    route.GetNearestTurn(distance, turn);
     size_t currentTurnIndex = 3;
     TEST(base::AlmostEqualAbs(distance,
                               mercator::DistanceOnEarth(pos, kTestGeometry[currentTurnIndex]), 0.1), ());
     TEST_EQUAL(turn, kTestTurns[currentTurnIndex - 1], ());
+
+    double timePassed = 0.8 * kTestTimes[2 - 1] + 0.2 * kTestTimes[3 - 1];
+
+    double time = route.GetCurrentTimeToEndSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[4] - timePassed, 0.1), ());
+
+    time = route.GetCurrentTimeToNearestTurnSec();
+    TEST(base::AlmostEqualAbs(time, kTestTimes[currentTurnIndex - 1] - timePassed, 0.1), ());
```

Comment:
TEST(..., (time, timePassed)); prints time value id test failed.

Reply:
Changed to TEST_ALMOST_EQUAL_ABS.